bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/util/StringUtils.java

```diff
@@ -0,0 +1,238 @@
+                    return objectValue.call(Scheduler.getStrand(), TO_STRING).toString();
+                }
+            }
+        }
+
+        BValue bValue = (BValue) value;
+        return bValue.stringValue(parent);
+    }
+
+    /**
+     * Returns the string value of Ballerina values in expression style.
+     *
+     * @param value The value on which the function is invoked
+     * @param parent The link to the parent node
+     * @return String value of the value in expression style
+     */
+    public static String getExpressionStringVal(Object value, BLink parent) {
+        if (value == null) {
+            return "()";
+        }
+
+        Type type = TypeUtils.getReferredType(TypeChecker.getType(value));
+
+        if (type.getTag() == TypeTags.STRING_TAG) {
+            return "\"" + ((BString) value).getValue() + "\"";
+        }
+
+        if (type.getTag() == TypeTags.DECIMAL_TAG) {
+            DecimalValue decimalValue = (DecimalValue) value;
+            return decimalValue.expressionStringValue(parent);
+        }
+
+        if (type.getTag() == TypeTags.FLOAT_TAG) {
+            if (Double.isNaN((Double) value)) {
+                return "float:" + value;
+            }
+            if (Double.isInfinite((Double) value)) {
+                return "float:" + value;
+            }
+        }
+
+        if (type.getTag() < TypeTags.NULL_TAG) {
+            return String.valueOf(value);
+        }
+
+        CycleUtils.Node node = new CycleUtils.Node(value, parent);
+
+        if (node.hasCyclesSoFar()) {
+            return STR_CYCLE + "[" + node.getIndex() + "]";
+        }
+
+        if (type.getTag() == TypeTags.MAP_TAG || type.getTag() == TypeTags.RECORD_TYPE_TAG) {
+            MapValueImpl mapValue = (MapValueImpl) value;
+            return mapValue.expressionStringValue(parent);
+        }
+
+        if (type.getTag() == TypeTags.ARRAY_TAG || type.getTag() == TypeTags.TUPLE_TAG) {
+            ArrayValue arrayValue = (ArrayValue) value;
+            return arrayValue.expressionStringValue(parent);
+        }
+
+        if (type.getTag() == TypeTags.TABLE_TAG) {
+            return ((BRefValue) value).expressionStringValue(parent);
+        }
+
+        if (type.getTag() == TypeTags.OBJECT_TYPE_TAG) {
+            AbstractObjectValue objectValue = (AbstractObjectValue) value;
+            ObjectType objectType = (ObjectType) TypeUtils.getReferredType(objectValue.getType());
+            for (MethodType func : objectType.getMethods()) {
+                if (func.getName().equals(TO_STRING) && func.getParameters().length == 0 &&
+                        func.getType().getReturnType().getTag() == TypeTags.STRING_TAG) {
+                    return "object " + objectValue.call(Scheduler.getStrand(), TO_STRING).toString();
+                }
+            }
+        }
+
+        BRefValue refValue = (BRefValue) value;
+        return refValue.expressionStringValue(parent);
+    }
```

Comment:
Shall we remove unnecessary white spaces? better to follow the conventions https://www.oracle.com/java/technologies/javase/codeconventions-whitespace.html

Reply:
I think these blank lines improve the code readability. https://www.oracle.com/java/technologies/javase/codeconventions-whitespace.html does not depict cases where blank lines should not be used, but it says that blank lines improve readability by setting off sections of code that are logically related. Please correct me if I am wrong.

Reply:
Will remove the newlines as it is expected by the WSO2 code style practices.

Reply:
Fixed with c24a2fbd2b99c9dd6dc76d45237874379999be35.